frontend/public/components/marketplace/marketplace-subscribe.tsx

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+
+import { Firehose, LoadingBox, history } from '../utils';
+import { referenceForModel, K8sResourceKind, k8sUpdate, k8sCreate } from '../../module/k8s';
+import { SubscriptionModel, CatalogSourceConfigModel, OperatorGroupModel, PackageManifestModel } from '../../models';
+import { OperatorGroupKind, PackageManifestKind, ClusterServiceVersionLogo, SubscriptionKind, InstallPlanApproval } from '../operator-lifecycle-manager';
+import { OperatorGroupSelector } from '../operator-lifecycle-manager/operator-group';
+import { RadioGroup } from '../radio';
+import { MARKETPLACE_CSC_NAME } from './index';
+
+// TODO: Use `redux-form` instead of stateful component
+const withFormState = (Component) => {
+  /**
+   * Controlled component which holds form state (https://reactjs.org/docs/forms.html#controlled-components).
+   */
+  return class WithFormState extends React.Component {
+    static WrappedComponent = Component;
+
+    state = {
+      target: null,
+      updateChannel: null,
+      approval: InstallPlanApproval.Automatic,
+    };
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+      const updateChannel = !_.isEmpty(nextProps.packageManifest.data)
+        ? (nextProps.packageManifest.data.status.channels.find(ch => ch.name === nextProps.packageManifest.data.status.defaultChannel) || nextProps.packageManifest.data.status.channels[0]).name
```

Comment:
Same here if we can have `_.get()` instead of assuming packageManifest will be defined